module/app/preset/src/main/java/org/openbase/bco/app/preset/DeviceNotificationApp.kt

```diff
@@ -0,0 +1,148 @@
+import org.openbase.type.language.MultiLanguageTextType.MultiLanguageText
+import org.slf4j.LoggerFactory
+import java.time.Duration
+import java.util.*
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+/**
+ * An app that notifies users about devices with empty batteries or offline states.
+ */
+class DeviceNotificationApp : AbstractAppController() {
+
+    private val batteryPool = CustomUnitPool<BatteryData, BatteryRemote>()
+    private var task: ScheduledFuture<*>? = null
+
+    init {
+        batteryPool.init(
+            UnitFilter.newBuilder().setProperties(UnitConfig.newBuilder().setUnitType(UnitType.BATTERY)).build()
+        )
+    }
+
+    @Throws(CouldNotPerformException::class, InterruptedException::class)
+    override fun execute(activationState: ActivationState): ActionDescription =
+        activationState.responsibleAction.also {
+            batteryPool.activate()
+            task?.cancel(false)
+            task = GlobalScheduledExecutorService.scheduleWithFixedDelay(
+                ::checkDevices,
+                INITIAL_VALIDATION_DELAY.toMillis(),
+                VALIDATION_PERIOD.toMillis(),
+                TimeUnit.MILLISECONDS
+            )
+            LOGGER.trace(getLabel() + " is running.")
+        }
+
+    @Throws(InterruptedException::class, CouldNotPerformException::class)
+    override fun stop(activationState: ActivationState) =
+        super.stop(activationState).also {
+            batteryPool.deactivate()
+            task?.cancel(true)
+            LOGGER.trace(getLabel() + " has been stopped.")
+        }
+
+    fun checkDevices() {
+        try {
+            batteryPool.internalUnitList
+                .onEach { remote -> remote.waitForData() }
+                .filter { remote ->
+                    when (remote.data.batteryState.value) {
+                        BatteryState.State.LOW, BatteryState.State.CRITICAL, BatteryState.State.UNKNOWN -> true
+                        else -> false
+                    }
+                }
+                .map { remote ->
+                    val messageBuilder: UserMessage.Builder = UserMessage.newBuilder()
+                    val textBuilder: MultiLanguageText.Builder = MultiLanguageText.newBuilder()
+
+                    MultiLanguageTextProcessor.addMultiLanguageText(
+                        textBuilder,
+                        Locale.ENGLISH,
+                        "Battery level of ${
+                            LabelProcessor.getBestMatch(
+                                Locale.ENGLISH,
+                                remote.config.label
+                            )
+                        } is ${remote.data.batteryState.value}"
+                    )
+
+                    MultiLanguageTextProcessor.addMultiLanguageText(
+                        textBuilder,
+                        Locale.GERMAN,
+                        "Batteriezustand von ${
```

Comment:
I feel like `Batteriezustand ist CRITICAL` sounds like the battery is broken and not empty what do you think about `Batterieladung` or`Batterieniveau`?

Reply:
totally right!